src/linear_algebra/finite_dimensional.lean

```diff
@@ -850,6 +850,61 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma findim_eq_zero_iff_subsingleton [finite_dimensional K V] :
+  findim K V = 0 ↔ subsingleton V :=
+begin
+  rw [← not_iff_not, ← not_nontrivial_iff_subsingleton, not_not],
+  change findim K V ≠ 0 ↔ _,
+  rw ← pos_iff_ne_zero,
+  exact findim_pos_iff
+end
+
+lemma eq_zero_of_findim_eq_zero [finite_dimensional K V]
+  (hV : findim K V = 0) (x : V) : x = 0 :=
+begin
+  rw findim_eq_zero_iff_subsingleton at hV,
+  cases hV,
+  exact hV x 0,
```

Comment:
Or even shorter as:
```suggestion
  haveI := findim_eq_zero_iff_subsingleton.mp hV,
  exact subsingleton.elim x 0,
```

I'm not sure this lemma is worth having, given its proof is so short

Reply:
Tha problem with `subsingleton.elim` is that it does not see `hV` so we have to do a `haveI : subsingleton V := hV` first which is not much better I think

Reply:
I don't follow, does my example not work?

Reply:
It does not compile for me

Reply:
I am sorry, I miss read your proof

Reply:
I still think the lemma is worth having since the corresponding one for `dim` exists and it can be useful in many places